yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -1958,6 +2093,424 @@ func getSupportedVersionString(minimumVersionsFixedIn map[string]*ybversion.YBVe
 	return strings.Join(supportedVersions, ", ")
 }
 
+// checkPermissionsOnAllNodes checks permissions on primary and all validated replicas.
+// It reports which nodes have permission issues or missing pg_stat_statements.
+func checkPermissionsOnAllNodes() error {
```

Comment:
consider breaking down this function.
Also name it as checkAssessmentPermissionsOnAllNodes()